NumberConverter.java

```diff
@@ -0,0 +1,243 @@
+                 }
+             } catch (InputMismatchException e) {
+                 System.out.println(Constants.ErrorMessages.ERROR_INVALID_INPUT_BASE);
+                 sc.next();
+             }
+         }
+     }
+     /*
+      * Prompts the user for a number and validates the input against the specified base.
+      * @param input  Scanner object to read user input.
+      * @param base Base of the input number.
+      * @return Validated number input from the user.
+      */
+     private static String getNumberFromUser(Scanner sc, int base) {
+         while (true) {
+             System.out.printf(Constants.ENTER_NUMBER_MESSAGE, base);
+             String number = sc.nextLine().trim();
+             String processedNumber = number.replace(" ", "");
+             if (processedNumber.length() > 100) {
+                 System.out.println(Constants.ErrorMessages.ERROR_NUMBER_TOO_LONG);
+                 continue;
+             }
+             if (isValidNumber(processedNumber, base)) {
+                 return processedNumber;
+             } else {
+                 System.out.println(Constants.ErrorMessages.ERROR_INVALID_NUMBER + base + ". Please try again.");
+             }
+         }
+     }
+     /*
+      * Validates the provided number for the specified base.
+      * Disallows negative and decimal numbers.
+      * @param number The number as a string to validate.
+      * @param base   The base of the number.
+      * @return true if valid, false otherwise.
+      */
+     private static boolean isValidNumber(String number, int base) {
+         if (number.contains("-") || number.contains(".")) {
+             return false;
+         }
+         for (int i = 0; i < number.length(); i++) {
+             char c = number.charAt(i);
+             if ((c >= '0' && c <= '9' && (c - '0' >= base)) ||
+                     (c >= 'A' && c <= 'F' && (c - 'A' + 10 >= base)) ||
+                     (c >= 'a' && c <= 'f' && (c - 'a' + 10 >= base))) {
+                 return false;
+             }
+         }
+         return true;
+     }
+     /*
+      * Converts the given number from one base to decimal (base 10).
+      * @param number The input number as a string.
+      * @param base   The base of the input number.
+      * @return The decimal equivalent of the input number as BigInteger.
+      */
+     private static BigInteger convertToDecimal(String number, int base) {
+         BigInteger decimalValue = BigInteger.ZERO;
+         for (int i = 0; i < number.length(); i++) {
+             char digit = number.charAt(i);
+             int value;
+             if (digit >= '0' && digit <= '9') {
+                 value = digit - '0';
+             } else if (digit >= 'A' && digit <= 'F') {
+                 value = digit - 'A' + 10;
+             } else {
+                 value = digit - 'a' + 10;
+             }
+             decimalValue = decimalValue.multiply(BigInteger.valueOf(base));

```

Comment:
Do not use in-built functions. Create your own function for performing calculations.